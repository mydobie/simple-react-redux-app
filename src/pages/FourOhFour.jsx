// Page rendered when url doesn't match route in App.jsx

import React from 'react';
import { Container, Row, Col } from 'reactstrap';
// EXAMPLE: Include an image
import notFoundImage from '../images/page_not_found.svg';

const FourOhFour = (/* props */) => (
  <Container>
    <Row>
      <Col>
        <h1>Page not found</h1>
        <p>The page you requested could not be found.</p>
        {/* EXAMPLE: Inline CSS styles */}
        <p style={{ textAlign: 'center' }}>
          <img src={notFoundImage} className='App-logo' alt='' />
        </p>
      </Col>
    </Row>
  </Container>
);

export default FourOhFour;
