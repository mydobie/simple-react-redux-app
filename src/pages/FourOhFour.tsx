// Page rendered when url doesn't match route in App.tsx

import React, { ReactElement } from 'react';
import { Row, Col } from 'react-bootstrap';
import notFoundImage from '../images/page_not_found.svg';

// *** Main component ***
const FourOhFour = (/* props */): ReactElement => (
  <Row data-testid='404PageContainer'>
    <Col>
      <h1>Page not found</h1>
      <p>The page you requested could not be found.</p>
      <p style={{ textAlign: 'center' }}>
        <img src={notFoundImage} className='App-logo' alt='' />
      </p>
    </Col>
  </Row>
);

export default FourOhFour;
