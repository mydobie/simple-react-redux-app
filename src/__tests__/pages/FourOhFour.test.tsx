/* eslint-disable react/react-in-jsx-scope */

import { axe } from 'jest-axe';
import { render } from '@testing-library/react';
import FourOhFour from '../../pages/FourOhFour';

describe('404 tests', () => {
  test('Is accessible', async () => {
    const { container } = render(<FourOhFour />);
    const results = await axe(container);
    expect(results).toHaveNoViolations();
  });

  test('Snapshot test on initial load', () => {
    const { container } = render(<FourOhFour />);
    expect(container).toMatchSnapshot();
  });
});
