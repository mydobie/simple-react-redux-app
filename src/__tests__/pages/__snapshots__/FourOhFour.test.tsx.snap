// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`404 tests Snapshot test on initial load 1`] = `
<div>
  <div
    class="row"
    data-testid="404PageContainer"
  >
    <div
      class="col"
    >
      <h1>
        Page not found
      </h1>
      <p>
        The page you requested could not be found.
      </p>
      <p
        style="text-align: center;"
      >
        <img
          alt=""
          class="App-logo"
          src="page_not_found.svg"
        />
      </p>
    </div>
  </div>
</div>
`;
